### DISCUSSION TITLE
Duplicate default workflows

### DISCUSSION THREAD

**Discussion by tay1orjones** [community/community#36584] [Projects and Issues]
The existing static list of default workflows available is nice, however it would be beneficial if these could be duplicated to have multiple outcomes.

Here's some examples of how this could be useful:

### Item added to project
1. When `pull request` is added, Set `Status: In Review`
2. When `issue` is added, Set `Status: Backlog`

### Item reopened
1. When `pull request` is reopened, Set `Status: In Review`
2. When `issue` is reopened, Set `Status: Backlog`

### Auto-archive items (beta)
1. When `issue`, `is:closed reason:not planned`
2. When `issue, pull request`, `is: closed/merged last-updated:56days`

In all these cases, only one of these can be applied. If we could duplicate the workflows, this could be done with the existing limited parameters available within the workflow configuration.

**Comment by 2percentsilk** [comment]
Hey @tay1orjones - thanks for your feedback and we're looking to address this soon! We've heard from many folks similar sentiments. I'll be sure to reach out if you're interested in giving feedback on how this might appear in the UI! :)

**Reply by MattMcAdams** [reply]
It would be cool to "Add custom workflow" that lets you achieve any of the functionality the default workflows have. That would cover this specific use case, and it could be expanded as more workflow options and actions are supported. I'd be happy to do a rough mockup or something if it would help?

This is one of the biggest missing pieces for our team in the new projects system (That and mobile / touch support, shout out to the GitHub Mobile beta!)

**Reply by 2percentsilk** [reply]
@MattMcAdams we've got some of these mocks ourselves 👀 
Currently, we're looking at just letting you duplicate the existing workflows as a first step but look forward to more building block UI in the future 🚀

**Reply by tay1orjones** [reply]
> if you're interested in giving feedback on how this might appear in the UI!

@2percentsilk Totally! I'm happy to help.

**Reply by castarco** [reply]
@2percentsilk 

What's the status of this? I see that this was opened around 2 years ago and I'm still facing the same problem.

Of course I understand it might not be a priority for whatever reason, but it would be nice to know how far are we from having it.

**Comment by akib1689** [comment]
Is the feature is implemented yet? I am also in the same type of problem for setting my projects.

**Reply by phisco** [reply]
+1 here, I saw that documented, but couldn't find it: https://docs.github.com/en/issues/planning-and-tracking-with-projects/automating-your-project/adding-items-automatically#duplicating-the-auto-add-workflow

**Reply by phisco** [reply]
turns out only "Auto-add" default workflow can be duplicated, you have to enable it before to get access to the "duplicate workflow" option shown in the docs

**Reply by akib1689** [reply]
this is auto adding item how do i set the item's field like status to backlog if it is an issue or in review if it is a pr.

**Reply by phisco** [reply]
Unfortunately it looks like you can only rely on the other workflow setting the default status for added items... I find it really difficult to handle both issues and PRs in the same board given the current implementation

**Reply by keikoro** [reply]
> turns out only "Auto-add" default workflow can be duplicated, you have to enable it before to get access to the "duplicate workflow" option shown in the docs

Actually, you can only duplicate it if you haven't yet hit the limit for your account. The limits are listed in a table further up the page you linked.

Which is why I'd like to log the following more user-friendly solution as **sub request**: don't restrict duplication per se, restrict the number of _active_ workflows a user (type) is allowed.

Duplicates which exceed the no. of allowed workflows could simply be set to inactive on creation. Then, when a user goes to enable an excess workflow, automatically disable the oldest workflow of the same type. Put a warning about the limit and what's going to happen when they flip the switch somewhere in close proximity to the on/off button.

**Comment by jaclynjessup** [comment]


**Reply by phisco** [reply]
See https://github.com/community/community/discussions/36584#discussioncomment-5673814

**Comment by DivergentEuropeans** [comment]
@2percentsilk It's kind of insane that there's no solution for this yet. It's been 1.5 years! It should be simple to be able to duplicate default workflows, if we can already duplicate 1 type of workflow

**Comment by smythp** [comment]
What's going on with this? Just pulled up project workflows and was amazed at how primitive it was. I thought I just couldn't find the add button.

**Comment by ShaneOH** [comment]
@2percentsilk Seems crazy that this was abandoned when we had a thread discussing mockups of this in 2022. Is there a new PM at GitHub that this got handed off to?

**Comment by apintols** [comment]
Would like to bump this thread, this functionality not existing is a huge hinderance to having both PRs and Issues in one project. Being able to apply an automation when an issue is added to a project, then a different automation when a PR is added to a project would be hugely helpful.

**Comment by lucasrodes** [comment]
Very interested in this. I'd like to set different status to issues and PRs when added to the project.

**Comment by akib1689** [comment]
For those who are still finding an answer, We solved this problem with this approach.

We created a repo in our organization, which is used for miscellaneous purposes. and we wrote an workflow to add issue to the project. using `github cli`.
